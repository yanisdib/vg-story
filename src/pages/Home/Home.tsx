import { Article } from '../../interfaces';

import { useArticles } from '../../hooks/useArticles';

import { selectFeaturedArticles, selectLatestFeaturedArticles } from '../../services/redux/selectors/articles';

import withStatusHandler from '../../helpers/hoc/withStatusHandler';

import { FeaturedArticlesGrid, Hero } from '../../components';


function Home(): JSX.Element {
    const { articles, error, status } = useArticles();

    const featuredArticles = selectLatestFeaturedArticles({
        body: articles,
        error,
        status
    });

    const HeroWithStatusHandler = withStatusHandler<{ article: Article }>(Hero);
    const FeaturedArticlesGridWithStatusHandler = withStatusHandler<{ featuredArticles: Article[] }>(FeaturedArticlesGrid)


    return (
        <>
            <HeroWithStatusHandler
                error={error}
                status={status}
                {...{ article: featuredArticles[0] }}
            />
            <FeaturedArticlesGridWithStatusHandler
                error={error}
                status={status}
                {...{
                    featuredArticles: featuredArticles.slice(1)
                }}
            />
        </>
    );
}


export default Home;