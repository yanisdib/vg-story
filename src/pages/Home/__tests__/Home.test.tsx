import { render } from '@testing-library/react';

import Home from '../Home'


describe('Home component', () => {
    it('renders without crashing', () => {
        const { container } = render(<Home />);
        
        expect(container).toMatchSnapshot();
    });
});