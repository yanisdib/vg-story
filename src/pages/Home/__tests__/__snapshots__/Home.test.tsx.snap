// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders without crashing 1`] = `
<div>
  <div
    class="sc-gKXOVf hzNlBD"
    data-testid="loading-screen"
  >
    <div
      class="sc-iBkjds kvvntL"
    >
      <div
        class="sc-ftvSup ixhnnA"
      >
        <div
          class="sc-jSMfEi bOwVaB"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-gKXOVf hzNlBD"
    data-testid="loading-screen"
  >
    <div
      class="sc-iBkjds kvvntL"
    >
      <div
        class="sc-ftvSup ixhnnA"
      >
        <div
          class="sc-jSMfEi bOwVaB"
        />
      </div>
    </div>
  </div>
</div>
`;
